racket/collects/racket/contract/private/merge-cache.rkt

```diff
@@ -0,0 +1,69 @@
+#lang racket/base
+
+(provide define/merge-cache)
+
+(require (for-syntax racket/base))
+
+;; How to evict things from the cache after some length of time ...
+;; Which functions are safe to put in the cache?
+;; known-good-contract?
+;; log all the predicates used in the benchmark
+;; TODO: use a thread cell here ...
+
+;; weak hashtable never cleared
+;; if reach a threshold, throw away and start over
+;; limit cache size to 10, and use some LRU policy (sort ...) (priority queue)
```

Comment:
How did you come up with the constant 10?

Reply:
I think this comment is outdated and was referring to various experiments I tried when we first implemented the caching strategy. We don't ever clear out the cache anymore.